### ISSUE TITLE
Add cross validation support

### ISSUE THREAD

**Issue by tpankaj** [sauhaardac/pytorch-neural-z2color#2]
Add a mechanism for evaluating the cross validation loss on unseen data as the network is training.
